src/overlays/actors/ovl_Boss_Hakugin/z_boss_hakugin.c

```diff
@@ -46,342 +113,3727 @@ ActorInit Boss_Hakugin_InitVars = {
+    2, 4, 5, 6, 7, 8, 8, 8, 8, 7, 6, 5, 4, 2,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_Hakugin/func_80B0CAF0.s")
+static s32 sParentShadowBodyParts[GOHT_BODYPART_MAX] = {
+    GOHT_BODYPART_THORAX,                // GOHT_BODYPART_PELVIS
+    BODYPART_NONE,                       // GOHT_BODYPART_THORAX
+    GOHT_BODYPART_THORAX,                // GOHT_BODYPART_HEAD
+    GOHT_BODYPART_THORAX,                // GOHT_BODYPART_FRONT_RIGHT_UPPER_LEG
+    GOHT_BODYPART_FRONT_RIGHT_UPPER_LEG, // GOHT_BODYPART_FRONT_RIGHT_LOWER_LEG
+    GOHT_BODYPART_FRONT_RIGHT_LOWER_LEG, // GOHT_BODYPART_FRONT_RIGHT_HOOF
+    GOHT_BODYPART_THORAX,                // GOHT_BODYPART_FRONT_LEFT_UPPER_LEG
+    GOHT_BODYPART_FRONT_LEFT_UPPER_LEG,  // GOHT_BODYPART_FRONT_LEFT_LOWER_LEG
+    GOHT_BODYPART_FRONT_LEFT_LOWER_LEG,  // GOHT_BODYPART_FRONT_LEFT_HOOF
+    GOHT_BODYPART_PELVIS,                // GOHT_BODYPART_BACK_RIGHT_THIGH
+    GOHT_BODYPART_BACK_RIGHT_THIGH,      // GOHT_BODYPART_BACK_RIGHT_SHIN
+    GOHT_BODYPART_BACK_RIGHT_SHIN,       // GOHT_BODYPART_BACK_RIGHT_HOOF
+    GOHT_BODYPART_PELVIS,                // GOHT_BODYPART_BACK_LEFT_THIGH
+    GOHT_BODYPART_BACK_LEFT_THIGH,       // GOHT_BODYPART_BACK_LEFT_SHIN
+    GOHT_BODYPART_BACK_LEFT_SHIN         // GOHT_BODYPART_BACK_LEFT_HOOF
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_Hakugin/func_80B0CCD8.s")
+static u8 sShadowSizes[GOHT_BODYPART_MAX] = {
+    GOHT_SHADOW_SIZE_LARGE,       // GOHT_BODYPART_PELVIS
+    GOHT_SHADOW_SIZE_EXTRA_LARGE, // GOHT_BODYPART_THORAX
+    GOHT_SHADOW_SIZE_LARGE,       // GOHT_BODYPART_HEAD
+    GOHT_SHADOW_SIZE_MEDIUM,      // GOHT_BODYPART_FRONT_RIGHT_UPPER_LEG
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_FRONT_RIGHT_LOWER_LEG
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_FRONT_RIGHT_HOOF
+    GOHT_SHADOW_SIZE_MEDIUM,      // GOHT_BODYPART_FRONT_LEFT_UPPER_LEG
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_FRONT_LEFT_LOWER_LEG
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_FRONT_LEFT_HOOF
+    GOHT_SHADOW_SIZE_MEDIUM,      // GOHT_BODYPART_BACK_RIGHT_THIGH
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_BACK_RIGHT_SHIN
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_BACK_RIGHT_HOOF
+    GOHT_SHADOW_SIZE_MEDIUM,      // GOHT_BODYPART_BACK_LEFT_THIGH
+    GOHT_SHADOW_SIZE_SMALL,       // GOHT_BODYPART_BACK_LEFT_SHIN
+    GOHT_SHADOW_SIZE_SMALL        // GOHT_BODYPART_BACK_LEFT_HOOF
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_Hakugin/func_80B0CF24.s")
+void BossHakugin_FillShadowTex(BossHakugin* this, u8* tex, f32 weight) {
+    s32 index;
+    s32 i;
+    s32 baseX;
+    s32 baseY;
+    s32 addY;
+    s32 x;
+    s32 y = BODYPART_NONE;
+    Vec3f lerp;
+    Vec3f* parentBodyPartPos;
+    Vec3f* bodyPartPos;
+    Vec3f pos;
+    Vec3f startVec;
+
+    for (i = 0; i < GOHT_BODYPART_MAX; i++) {
+        if ((weight == 0.0f) || (y = sParentShadowBodyParts[i]) > BODYPART_NONE) {
```

Comment:
Is the `y = sParentShadowBodyParts[i]` a fake match? Is this `y` as a body part used anywhere?

Reply:
It gets used about 4 lines lower. `parentBodyPartPos = &this->bodyPartsPos[y];` note Odolowa has the same pattern.

Reply:
Ah, missed that. Thanks!

Reply:
I should note that you can solve this in Odolwa with a `continue`, as I mentioned here: https://discord.com/channels/688807550715560050/688851976746041391/1155281303831314483

This avoids having to do the assignment in the if-statement. At the time I believed I solved it in EnSda too using the same pattern, but I tried to do it again and I couldn't get a match. Maybe I was just making a mistake though. I think it's likely you can match Goht with a `continue` too, but I haven't tried. I also haven't tried a `continue` for Wrath's shadow either.

My preference would be for us to follow as-is for now to keep things simple; this PR is huge as it is. Once this and Majora are in, I can definitely try to match all of them with a `continue` if that's what people would prefer. What do you think?

Reply:
Yeah that's fair. I definitely like the continue version but we can take a crack at it in bulk after Majora

Reply:
Maybe worth adding a comment here so it is easier to find?

Reply:
Sure, I put comments in EnSda, Goht, and Odolwa that all say `// TODO: match with a continue`